analyses/compare-gistic/util/GISTIC-comparison-functions.R

```diff
@@ -0,0 +1,345 @@
+  #                                `del_genes.conf_90.txt` file for the
+  #                                 medulloblastoma histology
+  
+  # Run `get_genes_object` function on each of the files
+  cohort_genes_vector <- get_genes_object(cohort_genes_file)
+  lgat_genes_vector <- get_genes_object(lgat_genes_file)
+  hgat_genes_vector <- get_genes_object(hgat_genes_file)
+  medulloblastoma_genes_vector <- get_genes_object(medulloblastoma_genes_file)
+  
+  # Run `plot_venn_diagram` for each comparison case
+  lgat_venn <- plot_venn_diagram(cohort_genes_vector, lgat_genes_vector, "lgat_genes")
+  hgat_venn <- plot_venn_diagram(cohort_genes_vector, hgat_genes_vector, "hgat_genes")
+  medulloblastoma_venn <- plot_venn_diagram(cohort_genes_vector, medulloblastoma_genes_vector, "medulloblastoma_genes")
+  
+  # Save plots to list
+  venn_plot_list <- list(lgat_venn, hgat_venn, medulloblastoma_venn)
+  
+  # Return the plot list
+  return(venn_plot_list)
+  
+}
+
+#### Implemented in `02-GISTIC-tidy-data-prep.Rmd` ----------------------------
+
+prepare_gene_level_gistic <- function(all_lesions_file,
+                                      amp_genes_file,
+                                      del_genes_file,
+                                      output_filename) {
+  
+  # Given the file paths to GISTIC's `all_lesion.conf_90.txt`,
+  # `amp_genes.conf_90.txt`, and `del_genes.conf_90.txt` files,
+  # read in and tidy this data into a data.frame that contains
+  # the amplified/deleted genes, the detection peak, and
+  # the IDs of the samples in the corresponding peaks along with
+  # with their CN status call.
+  #
+  # Args:
+  #   all_lesions_file: file path to GISTIC's `all_lesion.conf_90.txt` file
+  #   amp_genes_file: file path to GISTIC's `amp_genes.conf_90.txt` file
+  #   del_genes_file: file path to GISTIC's `del_genes.conf_90.txt` file
+  #   output_filename: string to save the output file as
+  #
+  # Return:
+  #   final_df: data.frame with the relevant data from the `all_lesions`,
+  #             `amp_genes`, and `del_genes` files
+  
+  # Read in `all_lesions_file`
+  gistic_all_lesions_df <- data.table::fread(all_lesions_file,
+                                             data.table = FALSE)
+  
+  # Run `get_genes_object` function on `amp_genes` and `del_genes` files to get
+  # a data.frame with just the genes and their corresponding detection peak
+  amp_genes_df <- get_genes_object(amp_genes_file,
+                                   include_peak_info = TRUE)
+  
+  del_genes_df <- get_genes_object(del_genes_file,
+                                   include_peak_info = TRUE)
+  
+  # Bind the rows from the above data.frames into one data.frame
+  final_df <- rbind(amp_genes_df, del_genes_df)
+  
+  
+  # Wrangle the GISTIC all lesions data to be in a comparable format with our CN calls
+  gistic_all_lesions_df <- gistic_all_lesions_df %>%
+    dplyr::select(
+      -c(
+        "Descriptor",
+        "Peak Limits",
```

Comment:
GISTIC's spaces in all their column names kill me.

Reply:
❗️